odl/space/ntuples.py

```diff
@@ -119,12 +119,15 @@ def element(self, inp=None, data_ptr=None):
                 return self.element_type(self, arr)
         else:
             if data_ptr is None:
-                arr = np.array(inp, copy=False, dtype=self.dtype, ndmin=1)
-                if arr.shape != (self.size,):
-                    raise ValueError('expected input shape {}, got {}.'
-                                     ''.format((self.size,), arr.shape))
+                if inp in self:
+                    return inp
+                else:
+                    arr = np.array(inp, copy=False, dtype=self.dtype, ndmin=1)
```

Comment:
This does not guarantee a no-op right? It will just try hard not to copy, but will do it if there is no other way.